pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -1379,8 +1360,9 @@ var _ = Describe("Topology", func() {
 				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
 			}
 			ExpectApplied(ctx, env.Client, provisioner)
-			pod := ExpectProvisioned(ctx, env.Client, recorder, provisioningController, prov, test.UnschedulablePod(test.PodOptions{
-				ObjectMeta: metav1.ObjectMeta{Labels: labels},
+			pod := ExpectProvisioned(ctx, env.Client, cluster, recorder, provisioningController, prov, test.UnschedulablePod(test.PodOptions{
+				ObjectMeta:   metav1.ObjectMeta{Labels: labels},
+				NodeSelector: map[string]string{v1.LabelInstanceTypeStable: "single-pod-instance-type"},
```

Comment:
This test was relying on the fact that the provisioner was not aware of cluster state, causing future pods to bind on in flight capacity. This forces the in flight capacity to reject additional pods.